### DISCUSSION TITLE
add requirements not detected by `pyinstaller`

### DISCUSSION THREAD

**Discussion by amtoine** [pyinstaller/pyinstaller#7159] [Help] [answered]
hello there :wave: :yum: 

i'm trying to find a way to package my [remote for linux using a PS3 controller](https://github.com/amtoine/remote-PS3) to simply use `remote` from my shell and not (**1**) `cd` to the source, (**2**) activate the virtual environment and (**3**) run the application...

`pyinstaller` works fine for simple scripts i've tried out, but now, as i'm using `evdev` to control my PS3 controller, it says
```bash
> pyinstaller --onefile remote.py
76 INFO: PyInstaller: 5.5
76 INFO: Python: 3.10.8
78 INFO: Platform: Linux-5.19.13-arch1-1-x86_64-with-glibc2.36
79 INFO: UPX is not available.
79 INFO: Extending PYTHONPATH with paths
...
> ./dist/remote
Traceback (most recent call last):
  File "remote.py", line 7, in <module>
    import evdev
ModuleNotFoundError: No module named 'evdev'
[13598] Failed to execute script 'remote' due to unhandled exception!
```

i could not find instructions on how to do that...
**is `evdev` installed but somehow not accessible from `./dist/remote`?**
**is `evdev` not supported by `pyinstaller` in some way?**

would love to know :star_struck:

**Comment by bwoodsend** [comment, marked as answer]
It works fine for me. If PyInstaller is not finding top level modules, that normally means that you're either running PyInstaller from the wrong environment or you've installed `evdev` into the wrong environment.

**Reply by amtoine** [reply]
thanks for the information :relieved: 

i tried with `pyinstaller` and all the requirements in the same venv and `evdev` is `import`ed at runtime :star_struck: 
some other issues now, but at least the `import`s work :muscle:
